### DISCUSSION TITLE
How to set local embedding model or use OpenAI key user provide to support files upload?

### DISCUSSION THREAD

**Discussion by piwawa** [danny-avila/LibreChat#3345] [Q&A]
I found that I can only write `RAG_OPENAI_API_KEY` in the .env file, but not automatically use the openai key inputed by the user like `OPENAI_API_KEY=user_provided`. This will result in an error message when the user uploads the document: `Error processing file` when i didn't set it. If I fill it in, my own key will be used for embedding. How can I automatically call the openai key entered by the user?

**Comment by piwawa** [comment]
?

**Comment by hayschan** [comment]
Does this documentation help you? 

https://www.librechat.ai/docs/configuration/rag_api#default-configuration

**Reply by piwawa** [reply]
> Does this documentation help you?
> 
> https://www.librechat.ai/docs/configuration/rag_api#default-configuration

I have read this POOR document before asking the question. This document is written ambiguously. What should I fill in? I tried set it like this:
```
RAG_OPENAI_API_KEY=user_provided
EMBEDDINGS_MODEL=text-embedding-3-small
```
But it throws an error directly and doesn't let me fill in the key!

![image](https://github.com/user-attachments/assets/8a367eb0-7753-45e0-90a2-97ffb48e1a7c)

Only setting my key in `.env`, it can work. Why can't just use the key what openai key user provided??
